src/linpred.jl

```diff
@@ -185,9 +195,10 @@ function SparsePredChol(X::SparseMatrixCSC{T}) where T
         similar(X))
 end
 
-cholpred(X::SparseMatrixCSC) = SparsePredChol(X)
+cholpred(X::SparseMatrixCSC, pivot::Bool=false) = SparsePredChol(X)
```

Comment:
Shouldn't the `pivot` argument be used here? It seems like this would avoid pivoting any sparse `X`, no?

Reply:
That makes sense to me. Unfortunately, I don't know enough about sparse matrices in Juia to know how to pivot them. If you can point me to a reference, I can try to add this.

Reply:
(Most of this pull request is built on the previous work which implemented this functionality for linear models. That author did not address sparse matrices, so I'm not sure what I need to do to add support for pivoting sparse matrices.)

Reply:
I think it should follow a similar approach to that of the `DensePredChol(X, pivot)` constructor, but make sure not to make the `X` matrix dense.
If you don't feel up to the challenge, maybe throw an error indicating this is not implemented yet when `cholpred(X::SparseMatrixCSC, pivot)` is called with `pivot=true`.

Reply:
I've updated the code to throw an error when `cholpred(X::SparseMatrixCSC, pivot)` is called with `pivot=true`. Maybe at some point in the future I'll attempt this in a different PR.